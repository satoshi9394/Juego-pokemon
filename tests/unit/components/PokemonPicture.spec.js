import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture';


describe('PokemonPicture component', () => {

  test('debe de hacer match con el snapshop', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: false
      }
    });
    expect(wrapper.html()).toMatchSnapshot() 
  });

  test('debe de mostrar la imagen oculta y el pokemon 100', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false
      }
    });
    const imgs = wrapper.findAll('img');
    expect(imgs[0].exists()).toBeTruthy();
    expect(imgs[1]).toBe(undefined);
    expect(imgs[0].classes('hidden-pokemon')).toBeTruthy();

  });

  test('debe de mostrar el pokemon si showPokemon: true', () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 1,
        showPokemon: true
      }
    });
    const img = wrapper.find('img');
    expect(img.exists()).toBeTruthy();
    expect(img.classes('fade-in')).toBeTruthy();
  });

});