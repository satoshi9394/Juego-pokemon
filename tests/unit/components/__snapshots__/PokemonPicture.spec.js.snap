// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component debe de hacer match con el snapshop 1`] = `
<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon">
  <!--v-if-->
</div>
`;
